blueman/main/PluginManager.py

```diff
@@ -205,28 +205,25 @@ def get_plugins(self):
     def Run(self, func, *args, **kwargs):
         rets = []
         for inst in self.__plugins.values():
-            try:
-                ret = getattr(inst, func)(*args, **kwargs)
-                rets.append(ret)
-            except Exception:
-                logging.error("Function %s on %s failed" % (func, inst.__class__.__name__), exc_info=True)
+            if hasattr(inst, func):
+                try:
+                    rets.append(getattr(inst, func)(*args, **kwargs))
+                except:
+                    logging.error("Function %s on %s failed" % (func, inst.__class__.__name__), exc_info=True)
 
         return rets
 
     #executes a function on all plugin instances, runs a callback after each plugin returns something
     def RunEx(self, func, callback, *args, **kwargs):
         for inst in self.__plugins.values():
-            ret = getattr(inst, func)(*args, **kwargs)
-            try:
-                ret = callback(inst, ret)
-            except StopException:
-                return ret
-            except Exception:
-                logging.error("Function %s on %s failed" % (func, inst.__class__.__name__), exc_info=True)
-                return
-
-            if ret is not None:
-                args = ret
```

Comment:
This was obviously meant for chaining. If the callback returns something it shall be used as `*args` for the next method call. I don't know if we actually use that anywhere, though.

Reply:
No, none of the callbacks passed to `RunEx` returns anything. They all just raise `StopException`s.

Reply:
Ah, good one, I missed that.